lib/Parse/ParseDecl.cpp

```diff
@@ -1485,6 +1481,12 @@ bool Parser::parseDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc) {
     return false;
   }
 
+  if (DK == DAK_Count && Tok.getText() == "NSKeyedArchiverClassName") {
+    DK = DAK_ObjC;
+    diagnose(Tok, diag::attr_nskeyedarchiverclassname_removed)
+      .fixItReplace(Tok.getLoc(), "objc");
```

Comment:
The Fix-It doesn't really work, because `@NSKeyedArchiverClassName` had quotes around the name but `@objc` doesn't, but... I really don't care because "nobody" is using `@NSKeyedArchiverClassName`.